calico-vpp-agent/watchers/pool_sync.go

```diff
@@ -13,7 +13,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package main
+package watchers
```

Comment:
This could be done by listening to the events generated by the ipam watcher (e.g. `IpamConfChanged`) in fact it could even be done as part of the `uplink_route_watcher` and that way we could get rid of the whole locking logic